packages/components/src/menu-item/index.tsx

```diff
@@ -15,8 +15,12 @@ import { cloneElement, forwardRef } from '@wordpress/element';
 import Shortcut from '../shortcut';
 import Button from '../button';
 import Icon from '../icon';
+import type { MenuItemProps } from './types';
 
-export function MenuItem( props, ref ) {
+export function MenuItem(
+	props: MenuItemProps,
```

Comment:
```suggestion
	props: WordPressComponentProps< MenuItemProps, 'button', false >,
```

Since we're passing down all the rest props to the underlying `button`, let's make sure we allow all the standard `button` props as well, like `onClick`. The `WordPressComponentProps` is a utility type to mix in those standard `button` props.

Reply:
Thanks for pointing this out 👍 , I should have missed those tips from the [Contributing guidance](https://github.com/WordPress/gutenberg/blob/25215a997e23b8b4b05b122ccb1c2778dcd4cfa7/packages/components/CONTRIBUTING.md#refactoring-a-component-to-typescript)

> Extend existing components’ props if possible, especially when a component internally forwards its props to another component in the package.
> If the component forwards its ...restProps to an underlying element/component, you should use the WordPressComponentProps type for the component's props:

Reply:
No worries, there's a lot of information in there 😆